### PR TITLE
updated link to ansible.builtin.user password

### PR DESCRIPTION
**Pull request by lmalivert**
##### SUMMARY

Update to link is in response to chat in community IRC channel stating that the link was not working. 

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

[lib/ansible/modules/user.py](https://github.com/ansible/ansible/compare/devel...lmalivert:ansible.builtin.user?expand=1#diff-f7246a0d1f290a8bdeaf4a775c920ed067d991cba363decc5ef3db4280e232d4 )
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by acozine** [comment]
Congratulations @lmalivert, you uncovered a bug! I built this branch to the test site, and there was no link at all. So I did some digging, and here's what I found:

The link reported on IRC is broken in the `latest` docs: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/user_module.html#parameter-password
The links works in the `devel` docs, but it's still a hard-coded link:
https://docs.ansible.com/ansible/devel/collections/ansible/builtin/user_module.html#parameter-password
(The hard-coded link was corrected in #73353 but the fix was not backported to the stable branch.)

You made exactly the right change to make this a sphinx link instead of a hard-coded HTML link, but it didn't work . . . and our own [documentation of how to do this](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation) is wrong. Because of how the tables are constructed in the module documentation, the `R(text,anchor)` syntax doesn't work. We need to use the clunkier `L(text,https://hard-coded-link.html)` syntax instead.

**Review by acozine — COMMENTED** [review]
See my earlier comment about uncovering a bug with this PR. I think my suggestion will make the link work with slightly better-looking text. Let's build to test to be sure before we merge.

File: lib/ansible/modules/user.py [hunk 1]
```diff
@@ -89,7 +89,7 @@
             - On macOS systems, this value has to be cleartext. Beware of security issues.
             - To create a disabled account on Linux systems, set this to C('!') or C('*').
             - To create a disabled account on OpenBSD, set this to C('*************').
-            - See U(https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-generate-encrypted-passwords-for-the-user-module)
+            - See R(user password,user_passwords)
```
Comment (acozine) [inline lib/ansible/modules/user.py]:
```suggestion
            - See L(this FAQ entry,https://docs.ansible.com/ansible/latest/reference_appendices/faq.html#how-do-i-generate-encrypted-passwords-for-the-user-module)
```

**Review by acozine — COMMENTED** [review]
See my earlier comment about uncovering a bug with this PR. I think my suggestion will make the link work with slightly better-looking text. Let's build to test to be sure before we merge.

File: lib/ansible/modules/user.py (hunk 1 above)
Comment (lmalivert) [reply lib/ansible/modules/user.py]: @acozine Oh wow. Okay, thanks Alicia!

Comment (samccann) [reply lib/ansible/modules/user.py]: where are we with this one? Someone else also opened a PR to fix it so it seems a common problem users are seeing

**Pull request merged**